### PR TITLE
Craft desktop View

### LINKED ISSUE #74: Desktop view
### REQUIREMENTS / DESCRIPTION

The current design of the application follows a mobile-first approach, but now it's necessary to adapt it to provide a responsive desktop view. This adaptation involves transforming the existing header and footer into a single sidebar layout, aligning with the provided Figma design.

#### Functionalities:
- Transform the existing header and footer elements into a single sidebar component.
- Ensure responsiveness across various desktop screen sizes.
- Maintain functionality of header and footer elements within the new sidebar layout.
- Implement the sidebar to collapse/expand functionality for better user experience, if necessary.

#### User Interaction:
- Users should be able to access all header and footer functionalities through the new sidebar.
- Ensure intuitive navigation within the sidebar, considering user habits and expectations.

#### Design Considerations:
- Follow the provided Figma design closely to maintain consistency with the overall application layout.
- Ensure that the sidebar design is visually appealing and coherent with the existing design language.
- Pay attention to spacing, typography, and color schemes to provide a seamless user experience.

#### Implementation Notes:
- Utilize CSS media queries to adjust the layout for different screen sizes.
- Consider CSS Grid or Flexbox for efficient layout management.
- Test the responsiveness across various desktop resolutions and browsers to ensure compatibility.
- Collaborate with UI/UX designers to address any design discrepancies or usability concerns.

#### Figma Sources:
image
image

### PR DESCRIPTION
**Pull request by Zama15**
## Desktop view
- make a button class for omni Auth google in the button components, 
- on button component were add different font size for the buttons on desktop view
- add different font sizes for inputs on the inputs components file
- add a close clickable button to close modal on modal component
- modify font sizes for the modal component
- remove the overflow to the posts cards container
- add a label for each section on the footer, the label was made with the class already implemented and a ::after selector, this label is only visible on desktop view
- move the config icon on the header to the top right on desktop view
- add the search view style on the style, the desktop and the mobile view
- add attributes to the main element on the feed file, this elements will contain the yields on the application file
- make a template with the grid attribute on the container class, this will modify the structure of the header and footer to achieve the desire desktop view from the Figma mockups 
- modify the stats on the profile view to not broke on desktop view

## View bugs
- if a user that is not the owner of the post tried to load the view it will be a redirect to the root path, posts_controller file were modified
- if a user tried search himself and see the profile before will render the user/show template, the users_controller were modified to load the posts/index instead which is the user logged profile

## Modfied stimulus controllers
- modal controller were modified from query selector and get elements to statics targets, and from event listeners to controllers actions on click
- profile controller were modified, the active class were bug not showing the correct state of the user from posts to photos and viceversa
- theme controller now show the theme will be change on click with a text

issue #74 

log in
image
sign up
image
feed
image
search
image
routines
image
profile
image
image
edit profile
image

thanks to @juperameza to help me with the stimulus controllers and the fixed routes on some controllers

### CODE REVIEW

**Review by PudinDeveloper — APPROVED** [review]

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -60,4 +61,9 @@ def get_name(user1, user2)
     user = [user1, user2].sort
     "private_#{user[0].id}_#{user[1].id}"
   end
+
+  def show_themself
+    @user = User.find(params[:id])
+    redirect_to posts_path if current_user.id == @user.id
+  end
```
Comment (juperameza) [inline app/controllers/users_controller.rb]: Consider using a scope for this or something

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by MauricioNC — APPROVED** [review]

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Zama15) [reply app/controllers/users_controller.rb]: okay, i just avoid make the query, deleting line 66 and comparing 
```
current.id == params[:id].to_i
```

**Review by BlazeDex — APPROVED** [review]

**Review by angel23v — APPROVED** [review]
If you make "commits" cleaner it will be nice because they are so many files for a PR

But nice keep working hard

 **REMEMBER WE ARE FAMILY**

**Review by juperameza — CHANGES_REQUESTED** [review]
Write what you did in this pr
I'm not clicking that issue 👺👺

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (juperameza) [reply app/controllers/users_controller.rb]: Ok

**Comment by Zama15** [comment]
> Write what you did in this pr I'm not clicking that issue 👺👺

my apologies for the late response

**Comment by Zama15** [comment]
> If you make "commits" cleaner it will be nice because they are so many files for a PR
> 
> But nice keep working hard
> 
> **REMEMBER WE ARE FAMILY**

yeah i think so, i will tried change this bad habit of mine :sob:

**Review by juperameza — APPROVED** [review]

**Pull request merged**